src/index.ts

```diff
@@ -1,60 +1,100 @@
-import * as core from '@actions/core';
-import * as exec from '@actions/exec';
-import * as io from '@actions/io';
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as io from "@actions/io";
 import * as fs from "fs";
 import * as path from "path";
 
-export async function run(): Promise<void> {
-    const imageInput = core.getInput('image', { required: true });
-    const tag = core.getInput('tag') || 'latest';
-    const registry = core.getInput('registry', { required: true });
-    const username = core.getInput('username', { required: true });
-    const password = core.getInput('password', { required: true });
-    const tlsVerify = core.getInput('tls-verify');
-    const digestFileInput = core.getInput('digestfile');
+interface Response {
+    exitCode: number;
+    stdout: string;
+    stderr: string;
+}
 
-    // get podman cli
-    const podman = await io.which('podman', true);
+async function run(): Promise<void> {
+    const imageInput = core.getInput("image", { required: true });
+    const tag = core.getInput("tag") || "latest";
+    const registry = core.getInput("registry", { required: true });
+    const username = core.getInput("username", { required: true });
+    const password = core.getInput("password", { required: true });
+    const tlsVerify = core.getInput("tls-verify");
+    const digestFileInput = core.getInput("digestfile");
 
-    const imageToPush = `${imageInput}:${tag}`;
-    let pushMsg = `Pushing ${imageToPush} to ${registry}`;
-    if (username) {
-        pushMsg += ` as ${username}`;
-    }
-    core.info(pushMsg);
+    // get Podman cli
+    const podman = await io.which("podman", true);
 
-    //check if images exist in podman's local storage
-    const checkImages = await execute(podman, ['images', '--format', 'json']);
+    let imageToPush = `${imageInput}:${tag}`;
 
-    const parsedCheckImages = JSON.parse(checkImages.stdout);
+    // check if image exist in Podman local registry
+    const isPresentInPodman: boolean = await checkImageInPodman(
+        imageToPush,
+        podman,
+    );
+
+    // check if image exist in Docker local registry and if exist pull the image to Podman
+    const isPresentInDocker: boolean = await pullImageFromDocker(
+        imageToPush,
+        podman,
+    );
 
-    // this is to temporarily solve an issue with the case-sensitive of the property field name. i.e it is Names or names??
-    const nameKeyMixedCase = parsedCheckImages[0] && Object.keys(parsedCheckImages[0]).find(key => 'names' === key.toLowerCase());
-    const imagesFound = parsedCheckImages.
-                            filter((image: string) => image[nameKeyMixedCase] && image[nameKeyMixedCase].find((name: string) => name.includes(`${imageToPush}`))).
-                            map((image: string ) => image[nameKeyMixedCase]);
+    // boolean value to check if pushed image is from Docker local registry
+    let isPushingDockerImage = false;
```

Comment:
If it's not present in docker _or_ podman, we could fail right here, since there's no point in execing a push we know will fail.